service/telemetry.go

```diff
@@ -98,7 +103,7 @@ func (tel *colTelemetry) initOnce(col *Collector) error {
 	instanceID := instanceUUID.String()
 
 	var pe http.Handler
-	if configtelemetry.UseOpenTelemetryForInternalMetrics {
+	if featuregate.IsEnabled(useOtelForInternalMetricsfeatureGateID) {
```

Comment:
Where is this gate registered?  I don't see it anywhere.  If there is a pre-existing gate in this package should it's `ID ` field be used directly here to ensure consistency?  See, [this example of registration](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/metrics_receiver_test.go#L38-L39) and [use](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/factory.go#L89).

Reply:
If I am understanding correctly of how feature gate should be used, the user should register the feature gate in their `init` function and set `Enabled` to `true` for this feature. 

I don't think I am understanding the example of registration you linked. How are those translated to feature flag?

Reply:
My apologies, I appear to have pasted the wrong link to the registration example.  It should be [this instead](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/config.go#L69-L77).

The code with the feature to be controlled by a feature gate is responsible for creating and registering the gate with the feature gate registry in an `init()` function.  The registry can then be manipulated by the user providing CLI flags with instructions to enable or disable a gated feature.  Finally, at runtime, the code with the feature to be controlled inquires whether the feature is enabled using the `featuregate.IsEnabled()` function.

In the present case, there should be something like this in `service/telemetry.go`:

```go
var useOtelForInternalMetrics = featuregate.Gate{
  ID: "telemetry.useOtelForInternalMetrics",
  Enabled: configtelemetry.UseOpenTelemetryForInternalMetrics,
  Description: "Controls whether to use the OTel SDK for internal metrics",
}

func init() {
  featuregate.Register(useOtelForInternalMetrics)
}
```

Reply:
ok I see. I had a different understanding of where registration should happen but i will add it